app/src/screens/StakingTokens.js

```diff
@@ -0,0 +1,218 @@
+import React, { useMemo } from 'react'
+import {
+  Box,
+  Distribution,
+  Field,
+  GU,
+  textStyle,
+  useTheme,
+  useViewport,
+} from '@aragon/ui'
+import { formatTokenAmount } from '../lib/token-utils'
+import { stakesPercentages } from '../lib/math-utils'
+
+const DISTRIBUTION_ITEMS_MAX = 6
+
+function displayedStakes(stakes, total) {
+  return stakesPercentages(
+    stakes.map(({ stakedAmount }) => stakedAmount),
+    {
+      total,
+      maxIncluded: DISTRIBUTION_ITEMS_MAX,
+    }
+  ).map((stake, index) => ({
+    item:
+      stake.index === -1
+        ? 'Others'
+        : `#${stakes[stake.index].proposal}${' '}${
+            stakes[stake.index].proposalName
+          }`,
+    percentage: stake.percentage,
+  }))
+}
+
+const StakingTokens = React.memo(function StakingTokens({
+  stakeToken,
+  myStakes,
+  myActiveTokens,
+  totalActiveTokens,
+}) {
+  const theme = useTheme()
+  const { below } = useViewport()
+  const compact = below('large')
+
+  const stakes = useMemo(() => {
+    if (!myStakes || !myActiveTokens) {
+      return
+    }
+    return displayedStakes(myStakes, myActiveTokens)
+  }, [myStakes, myActiveTokens])
+
+  const inactiveTokens = useMemo(() => {
+    if (!stakeToken.balance || !myActiveTokens) {
+      return
+    }
+    return stakeToken.balance - myActiveTokens
+  }, [stakeToken.balance, myActiveTokens])
+
+  return (
+    <Box heading="Staking tokens" padding={0}>
+      <div
+        css={`
+          border-bottom: 1px solid ${theme.border};
+        `}
+      >
+        <div
+          css={`
+            padding: ${3 * GU}px;
+          `}
+        >
+          <Field
+            label="Your tokens"
+            css={`
+              margin-bottom: 0px;
+            `}
+          >
+            <div
+              css={`
+                ${textStyle('title2')};
+              `}
+            >
+              {`${
+                stakeToken.balance
+                  ? formatTokenAmount(
+                      parseInt(stakeToken.balance),
+                      parseInt(stakeToken.tokenDecimals)
+                    )
+                  : '-'
+              } ${stakeToken.tokenSymbol}`}
+            </div>
+            <div
+              css={`
+                ${textStyle('body4')};
+                color: ${theme.contentSecondary};
+              `}
+            >
+              {stakeToken.balance
+                ? (parseInt(stakeToken.balance) /
+                    parseInt(stakeToken.tokenSupply)) *
+                  100
```

Comment:
Perhaps we can use a utility function for this ? 

We did one at https://github.com/1Hive/conviction-voting-app/pull/69/files#diff-53011e3821f05cfb30718a0380679c42, maybe we could use that one

Reply:
which one fabri? is taking me to a PR but not. sure which function you mean